convert_hf_to_gguf.py

```diff
@@ -2742,20 +2742,25 @@ def set_gguf_parameters(self):
         # ref: https://github.com/state-spaces/mamba/blob/ce59daea3a090d011d6476c6e5b97f6d58ddad8b/mamba_ssm/modules/mamba_simple.py#L58
         dt_rank      = self.find_hparam(["time_step_rank",     "dt_rank"],      optional=True) or -(d_model // -16)
         rms_norm_eps = self.find_hparam(["layer_norm_epsilon", "rms_norm_eps"], optional=True) or 1e-5
-
+        num_hidden_layers = self.find_hparam(["n_layer", "num_hidden_layers"])        
+        use_b_dt_norm = False
+        # For falconmamba we do apply RMS norm on B / DT and C layers
+        if self.find_hparam(["model_type"]) in ["falcon_mamba"]:
```

Comment:
This is not always present


```suggestion
        if self.find_hparam(["model_type"], optional=True) in ("falcon_mamba",):
```

Alternatively, `architectures` should always be there, because it's used to select the correct `Model` subclass


```suggestion
        if self.hparams["architectures"][0] in ("FalconMambaForCausalLM",):
```

*Or*, `FalconMambaModel` could be a subclass of `MambaModel` and simply override its `set_gguf_parameters` to add the metadata about the extra RMS norms.

Reply:
Thanks for the detailed answer, I'd go for option 1 !